serverscommands/instancecommands/rebuild.go

```diff
@@ -67,17 +71,36 @@ func commandRebuild(c *cli.Context) {
 	opts := osServers.RebuildOpts{
 		ImageID:    c.String("imageID"),
 		AdminPass:  c.String("adminPass"),
-		Name:       c.String("name"),
 		AccessIPv4: c.String("accessIPv4"),
 		AccessIPv6: c.String("accessIPv6"),
 	}
 
+	if c.IsSet("rename") {
+		opts.Name = c.String("rename")
+	}
+
 	if c.IsSet("metadata") {
 		opts.Metadata = util.CheckKVFlag(c, "metadata")
 	}
 
 	client := auth.NewClient("compute")
 	serverID := idOrName(c, client)
+
+	if c.IsSet("rename") {
+		opts.Name = c.String("rename")
+	} else if c.IsSet("id") { // Must get the name from compute by ID
+		getResult := osServers.Get(client, serverID)
```

Comment:
We can combine this line and the one below it. Also, the Gophercloud Rackspace `servers` package has a `Get` function: `serverResult, err := servers.Get(client, serverID).Extract()`